sky/serve/load_balancer.py

```diff
@@ -160,11 +160,12 @@ async def _proxy_with_retries(
         # SkyServe supports serving on Spot Instances. To avoid preemptions
         # during request handling, we add a retry here.
         retry_cnt = 0
+        failed_replica_urls: List[str] = []
         while True:
             retry_cnt += 1
             with self._client_pool_lock:
                 ready_replica_url = self._load_balancing_policy.select_replica(
-                    request)
+                    request, failed_replica_urls)
```

Comment:
Seems if this is the case, we will never retry for a transient network issue. How about we allow retrying on previously failed URLs if retry_cnt has not reached the maximum amount of retries?

If that cause too much overheads for retries, we can probably reduce the interval between retries?

Reply:
This is only retry logic in one request and it will still be selected by following requests. This PR is mainly for the case when the replica is in the `NOT_READY` state in the controller but not synced to LB yet; For the transient network, it will convert back to the `READY` state soon. Besides, if all possible replicas are in `failed_replica_urls`, then we will still choose from them.

Reply:
Oh, I am talking about a single request. Should we fail directly for that specific request if there are replicas available but just transient network issue during the time load balancer is sending that request to the replica? I think our original purpose for this retry is to retry on network issues. I am proposing the following to allow retries for the same replica if we have not reach max retry count yet.
```python
if ready_replica_url is None and failed_replica_urls:
    failed_replica_urls = []
```

Reply:
Fixed! PTAL again